framework/platform/bpl/uci/cfg/bpl_cfg_uci.cpp

```diff
@@ -162,6 +162,132 @@ int cfg_uci_get_wireless(enum paramType type, int index, const char param[], cha
     return status;
 }
 
+int cfg_uci_get_wireless_from_ifname(enum paramType type, char *interface_name, const char param[],
+                                     char *value)
+{
+    struct uci_context *ctx = uci_alloc_context();
+    if (!ctx) {
+        ERROR("%s, uci alloc context failed!\n", __FUNCTION__);
+        return RETURN_ERR;
+    }
+
+    char lookup_str[MAX_UCI_BUF_LEN] = "wireless";
+    struct uci_ptr ptr;
+    if ((uci_lookup_ptr(ctx, &ptr, lookup_str, true) != UCI_OK)) {
+        ERROR("%s, uci lookup failed!\n", __FUNCTION__);
+        uci_free_context(ctx);
+        return RETURN_ERR;
+    }
+
+    if (!ptr.p) {
```

Comment:
`ptr` should be initialized to `nullptr` and checked for validity here (mentioned this in one of the previous comments)

Reply:
the naming here is a bit confusing.
although named ptr, it is actually a statically defined struct with set of pointers and other information on the data:
```
struct uci_ptr
{
	enum uci_type target;
	enum {
		UCI_LOOKUP_DONE =     (1 << 0),
		UCI_LOOKUP_COMPLETE = (1 << 1),
		UCI_LOOKUP_EXTENDED = (1 << 2),
	} flags;

	struct uci_package *p;
	struct uci_section *s;
	struct uci_option *o;
	struct uci_element *last;

	const char *package;
	const char *section;
	const char *option;
	const char *value;
};
```